content/docs/rfcs/12/README.md

```diff
@@ -61,77 +63,151 @@ The following are not considered as part of the adversarial model:
 ## Protobuf
 
 ```protobuf
-message FilterRequest {
-  bool subscribe = 1;
-  string topic = 2;
-  repeated ContentFilter contentFilters = 3;
-
-  message ContentFilter {
-    string contentTopic = 1;
+syntax = "proto3";
+
+// 12/WAKU2-FILTER rfc: https://rfc.vac.dev/spec/12/
+package waku.filter.v2;
+
+// Protocol identifier: /vac/waku/filter-subscribe/2.0.0-beta1
+message FilterSubscribeRequest {
+  enum FilterSubscribeType {
+    SUBSCRIBER_PING = 0;
+    SUBSCRIBE = 1;
+    UNSUBSCRIBE = 2;
+    UNSUBSCRIBE_ALL = 3;
   }
+
+  string request_id = 1;
+  FilterSubscribeType filter_subscribe_type = 2;
+
+  // Filter criteria
+  optional string pubsub_topic = 10;
+  repeated string content_topics = 11;
 }
 
-message MessagePush {
-  repeated WakuMessage messages = 1;
+message FilterSubscribeResponse {
+  string request_id = 1;
+  uint32 status_code = 10;
```

Comment:
We should define the status codes, but in a separate recommendation spec so it can be extended/modified without having to change the spec when we modify/extend this list (specs should be as stable as possible and therefore the minimum set for a sensible implementation)